packages/v3-sdk/src/functions/getPool.ts

```diff
@@ -0,0 +1,54 @@
+import { Token } from '@pancakeswap/sdk'
+import { BigintIsh } from '@pancakeswap/swap-sdk-core'
+import invariant from 'tiny-invariant'
+import { FeeAmount } from '../constants'
+import { NoTickDataProvider, Tick, TickConstructorArgs, TickDataProvider, TickListDataProvider } from '../entities'
+
+export type PoolState = {
+  token0: Token
+  token1: Token
```

Comment:
Shall we use `Currency` instead? Considering the fact that we'll support native token pairs trading